python/lsst/pipe/base/graph/graph.py

```diff
@@ -356,27 +359,27 @@ def graph(self) -> nx.DiGraph:
 
     @property
     def inputQuanta(self) -> Iterable[QuantumNode]:
-        """Make a `list` of all `QuantumNode` objects that are 'input' nodes
-        to the graph, meaning those nodes to not depend on any other nodes in
+        """Return all `QuantumNode` objects that are 'input' nodes
+        to the graph, meaning those nodes do not depend on any other nodes in
         the graph.
 
         Returns
         -------
         inputNodes : iterable of `QuantumNode`
-            A list of nodes that are inputs to the graph
+            An iterable of nodes that are inputs to the graph
```

Comment:
Thanks for the fix, but no good deed goes unpunished - given that this a property, I think it should be documented like an attribute, i.e.
```
"""An iterable of nodes that are input to the graph (iterable of `QuantumNode`).

These are the nodes that do not depend on any other nodes in the graph.
"""
```

As an aside, having a `property` that's a generator expression bothers me a fair bit, because it means the iterator gets reset every time you access it, and that feels very un-attribute-like, but that's not something we can fix without an RFC.

Reply:
Unfortunately, it looks like _all_ the class properties have this problem... 😞

Reply:
Feel free to leave it, then.  I'll probably be doing some work in QG in the not-to-distant future myself and can deal with it then.

Reply:
I'm willing to clean up these docstrings myself. However, I'm not sure what to do about the types. For example, [allDatasetTypes](https://pipelines.lsst.io/v/daily/py-api/lsst.pipe.base.QuantumGraph.html#lsst.pipe.base.QuantumGraph.allDatasetTypes) returns a tuple of `lsst.pipe.base.graph._impleDetails.DatasetTypeName`, which, needless to say, is not a public type and does not appear in the Science Pipelines docs.

Reply:
@natelust, I'm guessing that's a NewType or something that should just appear as `str` in the docs?

Reply:
Oops, so it is. Never mind.